test/unit/lib/plugins/aws/package/compile/events/kafka.test.js

```diff
@@ -144,12 +149,41 @@ describe('test/unit/lib/plugins/aws/package/compile/events/kafka.test.js', () =>
       expect(eventSourceMappingResource.Properties).to.deep.equal(eventConfig.resource(awsNaming));
     };
 
-    it('should correctly compile EventSourceMapping resource properties for VPC_SECURITY_GROUP', async () => {
+    it('should fail to compile EventSourceMapping resource properties when no accessConfiguration supplied', async () => {
+      await expect(
+        runServerless({
+          fixture: 'function',
+          configExt: {
+            functions: {
+              basic: {
+                events: [
+                  {
+                    kafka: {
+                      topic,
+                      bootstrapServers: ['abc.xyz:9092'],
+                    },
+                  },
+                ],
+              },
+            },
+          },
+          command: 'package',
+        })
+      ).to.be.rejected.and.eventually.have.property(
+        'code',
+        'INVALID_NON_SCHEMA_COMPLIANT_CONFIGURATION'
```

Comment:
We usually don't write schema-specific tests as we would have to cover a ton, I would not change the approach here